app/src/main/java/com/example/bookwormadventuresdeluxe2/LoginActivity.java

```diff
@@ -288,16 +288,24 @@ public void onEvent(@Nullable QuerySnapshot queryDocumentSnapshots, @Nullable Fi
 
                                         default:
                                             /* Unexpected Error code*/
+                                            editTextEmail.setError(errorCode);
                                             throw new Exception("Unexpected Firebase Error Code"
                                                     + "inside click listener.");
                                     }
 
                                     /* Hide progress bar*/
                                     progressBar.setVisibility(View.INVISIBLE);
-                                } catch (Exception e)
+                                }
+                                catch (Exception e)
```

Comment:
Same, it is really weird to be throwing an exception then catching it 1 line down